impl/forms/AndeStarExtractorForms.h

```diff
@@ -0,0 +1,497 @@
+                                   const InstMetaData::PtrType &meta, InstMetaData::OperandTypes kind) const override
+    {
+        if (meta->isNoneOperandType(kind)) {
+            return 0;
+        } else if (meta->isAllOperandType(kind)) {
+            return getSourceRegs(icode);
+        } else {
+//THIS CODE NEEDS DETAILED TESTING -- See the form Form_C2_sp_store (yes store, it was easier to reference) C1 CI_sp
+            uint64_t result = 0;
+            //Using type of RS1 for the type of the implied GP aka X3
+            if (meta->isOperandType(InstMetaData::OperandFieldID::RS1, kind)) {
+                result |= (1ull << REGISTER_ANDES_GP);
+            }
+            return result;
+        }
+    }
+
+//004
+    uint64_t getDestOperTypeRegs(const Opcode icode,
+                                 const InstMetaData::PtrType &meta, InstMetaData::OperandTypes kind) const override
+    {
+        if (meta->isNoneOperandType(kind)) {
+            return 0;
+        } else if (meta->isAllOperandType(kind)) {
+            return getDestRegs(icode);
+        } else {
+            uint64_t result = 0;
+            if (meta->isOperandType(InstMetaData::OperandFieldID::RD, kind)) {
+                result |= extractUnmaskedIndexBit_(Form_AndeStar_Custom_0::idType::RD, icode, fixed_field_mask_);
+            }
+            return result;
+        }
+    }
+
+//005
+    OperandInfo getSourceOperandInfo(Opcode icode, const InstMetaData::PtrType& meta,
+                                     bool suppress_x0 = false) const override
+    {
+        OperandInfo olist;
+        //ADDIGP has implied global pointer, typ. X3, this is the only source
+        //using RS1 as the type index
+        olist.addElement(InstMetaData::OperandFieldID::RS1, meta->getOperandType(InstMetaData::OperandFieldID::RS1),
+                         REGISTER_ANDES_GP, false);
+        return olist;
+    }
+
+//006
+    OperandInfo getDestOperandInfo(Opcode icode, const InstMetaData::PtrType& meta,
+                                   bool suppress_x0 = false) const override
+    {
+        OperandInfo olist;
+        appendUnmaskedOperandInfo_(olist, icode, meta, InstMetaData::OperandFieldID::RD,
+                                   fixed_field_mask_, Form_AndeStar_Custom_0::idType::RD,
+                                   false, suppress_x0);
+        return olist;
+    }
+
+//007
+    ImmediateType getImmediateType() const override
+    {
+        return ImmediateType::SIGNED;
+    }
+
+//008
+    uint64_t getImmediate(const Opcode icode) const override
+    {
+        const uint64_t imm18 = extract_(Form_AndeStar_Custom_0::idType::IMM18, icode & ~fixed_field_mask_);
+        // Mapping:
+        // src    17  16:7   6   5:3    2:1   0
+        // dst    17  10:1  11  14:12  16:15  0
+        //                 11111111----------
+        //                 765432109876543210
+        return ((imm18 & 0b100000000000000000) >>  0)  //17    -> 17
```

Comment:
I think you could use the Swizzler here too if you wanted. But it may not end up being as easy to read as what you have here.